app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/ResurveyOpeningHours.kt

```diff
@@ -0,0 +1,78 @@
+    override val commitMessage = "resurvey opening hours"
+    override val icon = R.drawable.ic_quest_opening_hours
+    override fun getTitle(tags: Map<String, String>) = R.string.resurvey_opening_hours_title
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        return arrayOf(tags["name"]!!)
+    }
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox)) { element, geometry ->
+            if (element.tags != null) {
+                // require opening hours that are supported
+                if (OpeningHoursTagParser.parse(element.tags["opening_hours"]!!) != null) {
+                    handler.handle(element, geometry)
+                } else {
+                    Log.wtf("AAAA", "opening_hours=" + element.tags["opening_hours"] + " was rejected as not representable in SC")
+                }
+            }
+
+        }
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox): String {
+        val groupName = ".old_opening_hours"
+        val nodeGroupName = groupName + "_nodes"
+        val wayGroupName = groupName + "_ways"
+        val relationGroupName = groupName + "_relations"
+        val reviewIntervalInDays = 380
+        return bbox.toGlobalOverpassBBox() + "\n" +
+                getQueryPart("node", nodeGroupName, reviewIntervalInDays) +
+                getQueryPart("way", wayGroupName, reviewIntervalInDays) +
+                getQueryPart("relation", relationGroupName, reviewIntervalInDays) +
+                "($nodeGroupName; $wayGroupName; $relationGroupName;);\n" +
+                getQuestPrintStatement()
+    }
+
+    // Note that newer segment will ensure that any edit,
+    // including adding or updating review marker like check_date or survey:date tags
+    // will cause OSM elements to become ineligible for this quest for reviewIntervalInDays days.
+    // It allows supporting check_date and any other survey markers without parsing of any tags.
+    //
+    // objects where tag was surveyed and confirmed to be correct by adding check_date tags
+    // are with larger delay, as it means this info on this specific object is relatively stable
+    protected fun getQueryPart(objectType: String, nameOfGeneratedGroup: String, reviewIntervalInDays: Int) =
+            "$objectType[name][opening_hours]['opening_hours:signed'!='no'](if:!is_date(t['check_date:opening_hours']) || date(t['check_date:opening_hours']) < " +
+                    "date('${DateUtil.getOffsetDateString(-reviewIntervalInDays * 3)}T00:00:00Z')) -> .old_opening_hours_tag;\n" +
+                    "$objectType[name][opening_hours](newer: '${DateUtil.getOffsetDateString(-reviewIntervalInDays)}T00:00:00Z') -> .recently_edited_objects_with_opening_hours;\n" +
+                    "(.old_opening_hours_tag; - .recently_edited_objects_with_opening_hours;)-> $nameOfGeneratedGroup;\n"
```

Comment:
This would be slightly easier readable if you used a raw string (=multiline string).

Reply:
Good idea, changed.